src/dali_interface/dali_interface.py

```diff
@@ -38,6 +38,26 @@ class DaliFrame(NamedTuple):
     status: int = DaliStatus.OK
     message: str = "OK"
 
+    def __repr__(self):
+        if self.status != DaliStatus.OK:
+            return f"<DaliFrame {self.status}>"
+        data = "<DaliFrame "
+        if self.length == 8:
+            data = data + f"0x{self.data:02X}"
+        elif self.length == 16:
+            data = data + f"0x{self.data:04X}"
+        elif self.length == 24:
+            data = data + f"0x{self.data:06X}"
+        elif self.length == 32:
+            data = data + f"0x{self.data:08X}"
+        else:
+            data = data + f"0x{self.data:08X} (unknown length)"
+        if self.send_twice:
+            data = data + " send-twice"
+        if self.priority != 0:
+            data = data + f" priority: {self.priority}>"
+        return data
```

Comment:
(cleanup)

```suggestion
        if self.priority != 0: 
             data = data + f" priority: {self.priority}" 
         return data + ">"
```

However:

(personal preference) If I saw a repr without the priority, I'd assume it having the default priority. This is not the case, because default priority is 2. But I can also see why people may assume unstated = zero. I would nib this ambiguity in the bud by always stating the priority.

Reply:
Actually another rabbit hole opens up. `priority` makes only sense when a frame is send. At the moment, for received frames the priority is untouched and it will be the default value for sending frames.
I would only print out legal priorities, within [2,5] that is. And, in a second step make another PR where I ensure that priority is `None` for received frames.